lib/bundler/rubygems_integration.rb

```diff
@@ -67,6 +67,9 @@ def configuration
       Bundler.ui.error "#{e.class}: #{e.message}"
       Bundler.ui.trace e
       raise Gem::SystemExitException
+    rescue Psych::SyntaxError => e
+      raise YAMLSyntaxError.new(e, "Your RubyGems configuration file (usually located at ~/.gemrc) " \
```

Comment:
Unfortunately, `Psych::ParseError#file` is set to `nil`. And this error may be coming from different files:

* /etc/gemrc
* ~/.gemrc
* ENV['GEMRC']

Any ideas?